include/xgboost/base.h

```diff
@@ -232,7 +246,4 @@ using XGBoostVersionT = int32_t;
 #endif  // DMLC_USE_CXX11 && defined(__GNUC__) && !defined(__clang_version__)
 }  // namespace xgboost
 
-/* Always keep this #include at the bottom of xgboost/base.h */
-#include <xgboost/build_config.h>
```

Comment:
Can we include `xgboost/version_config.h` here? Otherwise, we'll have to fret about which header gets included first.

Reply:
@hcho3 I don't think version have any effect to the rest of the macros right?